rfcs/013-utility-first-css-framework.md

```diff
@@ -0,0 +1,276 @@
+<!-- - [Tailwind UI](https://tailwindui.com/)
+  - Component library built with TailwindCSS, Headless UI and Heroicons
+  - Requires a license. I have an individual license, but a team license is a one-time fee of $500 (discounted from $799 from already owning a license) which includes up to 25 seats
+  - The major benefit of something like Tailwind UI is that these components are not imported as a npm library, but rather copy/pasted from examples - so that we are then free to evolve and modify the component as we see fit - seems to be ideal for both getting ready made components for basic things - as well as not being constrained in the design language -->
+
+### Expected results
+
+- More consistent styling
+  - TailwindCSS by itself provides a predefined set of of font sizes, widths, breakpoints, colors (which all can be modified)
+- Higher overall quality of UI
+  - Having to write less custom CSS through TailwindCSS, and having to build fewer basic components ourselves by using Tailwind UI should result in a overall higher quality of components, including better accessibility
+- Increased efficiency of building and maintaining UI
+  - Less naming things, less writing fully custom css, less importing of packages (styled-components, styled-system), less importing of abstracted "utility" components (<Grid />, <Flex/>)
+  - Easier to maintain, read and understand existing UI code (with utility classes being present directly in the composition of html elements)
+- @opencollective/design and @opencollective/developers can focus less on making sure that basic components like Buttons, Dropdowns, Modals, Forms, etc, look good and function well - and focus more on domain-specific design challenges that are unique to Open Collective - and have access to more consistent primitives while doing so
+
+## Proof of concept
+
+[PR that adds TailwindCSS](https://github.com/opencollective/opencollective-frontend/pull/8827) and some additional changes including:
+
+- OC primary color added to tailwind config
+  - With a workaround to set the `primary` palette colors dynamically (for the Collective Theme) using styeld jsx in `DefaultPaletteStyle.tsx` used in `_app.js` and `CollectiveThemeProvider.js`, so that it can be with Tailwind
+- Some global styling removed from `app.css`
+  - Removed the global font-size rules (was set to 62.5%, and 55% on smaller screens)
+    - Tailwind CSS expects this to be default (16px)
+    - Replaced all `rem` values with a new value that is divided by 1.6 (and rounded to nearest 0.05 rem)
+
+## Adoption / Transition strategy
+
+Since there is a lot of components and UI built with styled-components, we'd have to accept that we would live with both libraries for a while. We could adopt Tailwind and the utilify-first suite as the new main approach to styling - while slowly building new components and replacing old stuff using the new suite - motivated by new and better looks, improved dx, and better accessibility for the end user.
```

Comment:
We should clarify our policy towards using styled-components: will it be (1) Discouraged for new developments (2) Forbidden for new components (3) Forbidden for all new code, even when working with legacy pages.

While (2) is probably what we want in the long term, I would suggest starting with (1) to make the transition smoother.

Reply:
Agree that (1) makes sense, added a note about this.